### PR TITLE
Improve ui pending state command info #395 

### LINKED ISSUE #395: UI: Improve pending state message
State 3 different pending state. New, Deleted, Edited. With countdown in secs at the right side of the state. Show state and spinner after countdown reaches zero.

### PR DESCRIPTION
**Pull request by YijinL**
First half of #395 

@yl-coder: now `ReadOnlyViewablePerson` has a `getChangeInProgress()` method which returns the enum:
```java
enum ChangeInProgress { NONE, ADDING, EDITING, DELETING }
```
You can also listen to it's `ReadOnlyProperty` from the `changeInProgress()` method.

### CODE REVIEW

File: src/main/java/address/model/AddPersonCommand.java [hunk 1]
```diff
@@ -82,7 +85,10 @@ protected void after() {
     protected State simulateResult() {
         assert input != null;
         // create VP and add to model
-        viewableToAdd = PlatformExecUtil.callAndWait(() -> model.addViewablePersonWithoutBacking(input), null);
+        PlatformExecUtil.runAndWait(() -> {
+            viewableToAdd = model.addViewablePersonWithoutBacking(input);
+            viewableToAdd.setChangeInProgress(ADDING);
```
Comment (yl-coder) [inline src/main/java/address/model/AddPersonCommand.java]: Is addViewablePersonWithoutBacking use in other places? If not, i think  viewableToAdd.setChangeInProgress(ADDING); should be inside that method.

**Comment by yl-coder** [comment]
Looks fine. no other comments.

File: src/main/java/address/model/AddPersonCommand.java (hunk 1 above)
Comment (YijinL) [inline src/main/java/address/model/AddPersonCommand.java]: No, but the status setting makes more sense existing in the command than the model code

**Comment by damithc** [comment]
Branch can be deleted.

**Pull request merged**